include/circt/Dialect/Calyx/CalyxStructure.td

```diff
@@ -117,3 +148,37 @@ def ControlOp : CalyxContainer<"control", []> {
     ```
   }];
 }
+
+def CellOp : CalyxOp<"cell", [
+    HasParent<"CellsOp">
+  ]> {
+  let summary = "Calyx Cell";
+  let description = [{
+    Represents a cell (or instance) of a Calyx component or
+    primitive, which may include state. Some cells may
+    optionally have parameters attributed to them.
+
+    ```mlir
+      %in1, %out1 = calyx.cell "name" @MyComponent : i64, i16
+    ```
+  }];
+
+  let extraClassDeclaration = [{
+    /// Lookup the component for the symbol. This returns null on
+    /// invalid IR.
+    Operation *getReferencedComponent();
```

Comment:
Same suggestion, I think this could return ComponentOp and save some casting later.